CRM/Activity/BAO/Activity.php

```diff
@@ -2413,6 +2413,9 @@ public static function checkPermission($activityId, $action) {
       if ($sourceContactId) {
         $allow = CRM_Contact_BAO_Contact_Permission::allow($sourceContactId, $permission);
       }
+      else {
```

Comment:
Yes , I always feel a lot of "elses" in the code increase complexity and make it harder to read. (It is just an opinion though :) )